src/Stack.elm

```diff
@@ -134,30 +134,54 @@ member sought (Stack _ xs) =
 -}
 append : Stack a -> Stack a -> Stack a
 append (Stack xsCount xs) (Stack ysCount ys) =
-  List.append xs ys 
+  List.append xs ys
     |> Stack (xsCount + ysCount)
 
 {-| Given a list of stacks, join them together where the top of the first
     `Stack` in the given list is the top of the returned `Stack`.
 
-      stacks = 
+      stacks =
         [ Stack.fromList [1, 2]
         , Stack.fromList [3, 4]
         , Stack.fromList [5, 6]
         ]
 
-      Stack.concat stacks      => Stack [1, 2, 3, 4, 5, 6]
+      Stack.concat stacks      => Stack 6 [1, 2, 3, 4, 5, 6]
 -}
 concat : List (Stack a) -> Stack a
-concat = List.foldr append empty 
+concat = List.foldr append empty
 
-{-| Given a function `(a -> b)` and a `Stack a` return a new 
+{-| Given a function `(a -> b)` and a `Stack a` return a new
     `Stack b` with the function applied to every element of the stack.
 
-      Stack.fromList [1, 2, 3] 
-        |> Stack.map ((+) 1)      => Stack [2, 3, 4]
+      Stack.fromList [1, 2, 3]
+        |> Stack.map ((+) 1)      => Stack 3 [2, 3, 4]
 -}
 map : (a -> b) -> Stack a -> Stack b
 map f (Stack count xs) =
   List.map f xs
     |> Stack count
+
+
+{-| Filter items within stack, similar to `List.filter`
+
+      Stack.fromList [1, 2, 3]
+        |> Stack.filter ((<) 1)      => Stack 2 [2, 3]
+
+-}
+filter : (a -> Bool) -> Stack a -> Stack a
+filter f (Stack _ xs) =
```

Comment:
I like this implementation, but there is an arguably more simple one:
```elm
filter : (a -> Bool) -> Stack a -> Stack a
filter predicate (Stack _ xs) =
  let
    newList = List.filter predicate xs 
  in
  Stack (List.length newList) newList
```

Did you consider performance in your implementation? It does look like [List.filter](https://github.com/elm/core/blob/master/src/List.elm#L213) and [List.length](https://github.com/elm/core/blob/master/src/List.elm#L250) use `foldr`, so your solution removes the need to do that twice which I like.

What are your thoughts?

Reply:
I did not try to measure the real performance implications. Both `length` and `filter` has to visit every item in the list at least once. With folder both of these are done together so in theory it should be faster. But I believe both implementation would be O(n) so scalability would not be concern in either case.

I've went for version which seems algorithmically faster because it was still faster to implement optimal version than trying to measure cost of `List.filter` + `List.length`.

Reply:
I'm cool with this implementation, I just wanted to hear your reasoning behind it!